### PR TITLE
Skip authentication in for raw content requetes if the bucket is public

### PR DESCRIPTION
**Pull request by marcocast**

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java [hunk 1]
```diff
@@ -140,7 +140,7 @@ public CatalogObjectMetadata get(
             @ApiParam(value = "sessionID", required = false) @RequestHeader(value = "sessionID", required = false) String sessionId,
             @PathVariable String bucketName, @PathVariable String name, @PathVariable long commitTimeRaw)
             throws UnsupportedEncodingException, NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [CatalogObjectController.java#L108](https://github.com/ow2-proactive/catalog/blob/c2a064eae04f06a81cbe31f5b5dc963494953fee/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java#L108): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [CatalogObjectRevisionController.java#L97](https://github.com/ow2-proactive/catalog/blob/c2a064eae04f06a81cbe31f5b5dc963494953fee/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java#L97): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [CatalogObjectRevisionController.java#L120](https://github.com/ow2-proactive/catalog/blob/c2a064eae04f06a81cbe31f5b5dc963494953fee/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java#L120): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [RestApiAccessService.java#L59](https://github.com/ow2-proactive/catalog/blob/c2a064eae04f06a81cbe31f5b5dc963494953fee/src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java#L59): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [RestApiAccessService.java#L70](https://github.com/ow2-proactive/catalog/blob/c2a064eae04f06a81cbe31f5b5dc963494953fee/src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java#L70): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [CatalogObjectController.java#L113](https://github.com/ow2-proactive/catalog/blob/8fdb7d2392ceae23a887ea14acb35f1df3471307/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java#L113): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [CatalogObjectRevisionController.java#L97](https://github.com/ow2-proactive/catalog/blob/8fdb7d2392ceae23a887ea14acb35f1df3471307/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java#L97): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [CatalogObjectRevisionController.java#L120](https://github.com/ow2-proactive/catalog/blob/8fdb7d2392ceae23a887ea14acb35f1df3471307/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java#L120): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [RestApiAccessService.java#L59](https://github.com/ow2-proactive/catalog/blob/8fdb7d2392ceae23a887ea14acb35f1df3471307/src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java#L59): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [RestApiAccessService.java#L70](https://github.com/ow2-proactive/catalog/blob/8fdb7d2392ceae23a887ea14acb35f1df3471307/src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java#L70): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by tobwiens** [comment]
You can remove this logic in class `AuthorizationService` 

```
if (bucketOwnerOrGroup == null || bucketOwnerOrGroup.equals(BucketService.DEFAULT_BUCKET_OWNER)) {
            return true;
        }
```

And

File: src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java [hunk 2]
```diff
@@ -104,9 +104,9 @@ public BucketMetadata create(
     public BucketMetadata getMetadata(
             @SuppressWarnings("DefaultAnnotationParam") @ApiParam(value = "sessionID", required = false) @RequestHeader(value = "sessionID", required = false) String sessionId,
             @PathVariable String bucketName) throws NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java [hunk 3]
```diff
@@ -153,9 +153,9 @@ public void cleanEmpty() {
     public BucketMetadata delete(
             @ApiParam(value = "sessionID", required = false) @RequestHeader(value = "sessionID", required = false) String sessionId,
             @PathVariable String bucketName) throws NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java [hunk 4]
```diff
@@ -95,9 +95,9 @@ public CatalogObjectMetadata create(
             @ApiParam(value = "The commit message of the CatalogRawObject Revision", required = true) @RequestParam String commitMessage,
             @RequestPart(value = "file") MultipartFile file)
             throws IOException, NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java [hunk 5]
```diff
@@ -118,9 +118,9 @@ public CatalogObjectMetadata get(
             @ApiParam(value = "sessionID", required = false) @RequestHeader(value = "sessionID", required = false) String sessionId,
             @PathVariable String bucketName, @PathVariable String name, @PathVariable long commitTimeRaw)
             throws UnsupportedEncodingException, NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 6]
```diff
@@ -111,9 +111,9 @@ public CatalogObjectMetadataList create(
             @ApiParam(value = "The content type of CatalogRawObject - MIME type", required = true) @RequestParam String objectContentType,
             @ApiParam(value = "The content of CatalogRawObject", required = true) @RequestPart(value = "file") MultipartFile file)
             throws IOException, NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java [hunk 7]
```diff
@@ -55,15 +55,16 @@ public RestApiAccessService(BucketService bucketService, AuthorizationService au
...
 
-    public RestApiAccessResponse checkAccessBySessionIdForBucketAndThrowIfDeclined(String sessionId, String bucketName)
-            throws NotAuthenticatedException, AccessDeniedException {
-        RestApiAccessResponse restApiAccessResponse = this.checkAccessBySessionForBucketToOwnerOrGroup(sessionId,
-                                                                                                       bucketName);
-        if (!restApiAccessResponse.isAuthorized()) {
-            throw new AccessDeniedException("SessionId: " + sessionId + " is not allowed to access buckets with id " +
-                                            bucketName);
+    public void checkAccessBySessionIdForBucketAndThrowIfDeclined(boolean sessionIdRequired, String sessionId,
+            String bucketName) throws NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 13 issues
* ![MINOR][MINOR] 13 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [BucketController.java#L87](https://github.com/ow2-proactive/catalog/blob/b6ea8f4784379d4c8d92852549443245f84e8117/src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java#L87): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [BucketController.java#L123](https://github.com/ow2-proactive/catalog/blob/b6ea8f4784379d4c8d92852549443245f84e8117/src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java#L123): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [CatalogObjectReportController.java#L92](https://github.com/ow2-proactive/catalog/blob/b6ea8f4784379d4c8d92852549443245f84e8117/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectReportController.java#L92): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [CatalogObjectReportController.java#L149](https://github.com/ow2-proactive/catalog/blob/b6ea8f4784379d4c8d92852549443245f84e8117/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectReportController.java#L149): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [RestApiAccessService.java#L71](https://github.com/ow2-proactive/catalog/blob/b6ea8f4784379d4c8d92852549443245f84e8117/src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java#L71): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**